api/src/test/java/org/openmrs/module/pihcore/PihCoreContextSensitiveTest.java

```diff
@@ -4,22 +4,42 @@
 import org.apache.commons.io.IOUtils;
 import org.openmrs.EncounterType;
 import org.openmrs.api.context.Context;
+import org.openmrs.module.Module;
+import org.openmrs.module.ModuleFactory;
 import org.openmrs.module.initializer.Domain;
 import org.openmrs.module.initializer.InitializerConstants;
 import org.openmrs.module.initializer.api.ConfigDirUtil;
 import org.openmrs.module.initializer.api.InitializerService;
 import org.openmrs.module.pihcore.metadata.Metadata;
 import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.test.annotation.DirtiesContext;
 
 import java.io.File;
 import java.io.InputStream;
 import java.nio.file.Paths;
+import java.sql.SQLException;
 import java.util.Properties;
 
 import static org.openmrs.module.initializer.api.ConfigDirUtil.CHECKSUM_FILE_EXT;
 
+@DirtiesContext
 public abstract class PihCoreContextSensitiveTest extends BaseModuleContextSensitiveTest {
 
+    public PihCoreContextSensitiveTest() {
+        super();
+        {
+            Module mod = new Module("", "metadatamapping", "", "", "", "1.3.4");
+            mod.setFile(new File(""));
+            ModuleFactory.getStartedModulesMap().put(mod.getModuleId(), mod);
+        }
+    }
+
+    @Override
```

Comment:
I spent a really long time fighting unit tests today, and this is what I was able to come up with to get everything running and passing, unfortunately it really slows things down.  We can investigate and try to improve later on, but for now this works.  Basically I think we need to add the metadatamapping module like this to get Initializer to load in metadata mappings in the unit tests, but then doing so somehow messes up tests when run in the suite.  I was able to mostly fix that with the DirtiesContext annotation, but then a few tests started randomly failing and prompting user/password login prompts.  Adding the deleteAllData for all tests as below ended up fixing this, but this is pretty brute force.